Turbulence.ModelGenerator/Preprocessing.cs

```diff
@@ -0,0 +1,136 @@
+                var match = Regex.Match(line, paramsPattern);
+                if (match.Success && lastH2 != null)
+                {
+                    // Use the last header title instead of [...] Params
+                    name = Regex.Match(lastH2, @"## (.*) %.*").Groups[1].Value;
+
+                    if (line.Contains("Query String Params"))
+                    {
+                        name += "QueryParams";
+                    }
+                    else if (line.Contains("Response Body"))
+                    {
+                        name += "Response";
+                    }
+                    else
+                    {
+                        name += "Params";
+                    }
+                }
+                else
+                {
+                    // Markdown heading to pretty class name
+                    /* language=RegEx */
+                    const string pattern = @"###### (.*?)( Structure| Fields)?$";
+                    name = Regex.Match(line, pattern).Groups[1].Value;
+                }
+
+                name = name.Replace(" ", "")
+                           .Replace("/", "")
+                           .Replace("-", "");
+
+                string? table = await ExtractTable(reader);
+
+                // Not a valid table or not a table we are interested in, continue
+                if (table == null) continue;
+
+                // We use the name of the file for the directory name to store all the tables to
+                // We assume there are no duplicates
+                string dir = Regex.Replace(file, ".md$", "")
+                                  .Split(Path.DirectorySeparatorChar)[^1]
+                                  .Replace('_', ' ');
+                dir = CultureInfo.InvariantCulture.TextInfo.ToTitleCase(dir).Replace(" ", "");
+                
+                // Special case for OAuth2
+                if (dir == "Oauth2") dir = "OAuth2";
+                
+                dir = Path.Combine(outPath.AbsolutePath, dir);
+
+                string filename = Path.Combine(dir, name) + ".md";
+                
+                // Create directory recursively
+                Directory.CreateDirectory(dir);
+
+                // Throw error if we have already written to this file
+                if (writtenFiles.Contains(filename))
+                {
+                    throw new Exception($"Tried to write duplicate file {filename}.");
+                }
+
+                writtenFiles.Add(filename);
+
+                // Write table to file
+                await File.WriteAllTextAsync(filename , table);
+            }
+        }
+    }
+
+    private static async Task<string?> ExtractTable(TextReader reader)
+    {
+        StringBuilder table = new();
+        string? line;
+        
+        // Get rid of any whitespace lines
+        do
+        {
+            line = await reader.ReadLineAsync();
+        } while (line != null && Regex.Match(line, @"^\s*$").Success);
```

Comment:
Couldn't you just use ltrim here?

Reply:
LTrim removes whitespace, this reads new lines until there are no more whitespace lines